### DISCUSSION TITLE
TC-Gen: Issue counting miss events with init_hour filter applied

### DISCUSSION THREAD

**Discussion by halperin-erau** [dtcenter/METplus#1596] [Statistical Computation] [answered]
Hi all,

I've identified an issue with how TC-Gen counts miss events when the init_hour filter is applied.

Let's consider a case where a user wants to verify only the 00Z and 12Z model initialization cycles.  In the config file, they would set

`init_hour = ["00", "12"];`

I ran TC-Gen using the above init_hour filter on GFS forecast data over the Atlantic basin from the [use case](https://metplus.readthedocs.io/en/main_v4.1/generated/model_applications/tc_and_extra_tc/TCGen_fcstGFS_obsBDECK_2021season.html#sphx-glr-generated-model-applications-tc-and-extra-tc-tcgen-fcstgfs-obsbdeck-2021season-py).  The output shows that TC-Gen correctly verified genesis forecasts (which were classified as hits or false alarms) from only the 00Z and 12Z initialization cycles.  However, TC-Gen counted miss events from 00Z, 06Z, 12Z, and 18Z model initialization cycles.  I would suggest that miss events from 06Z and 18Z initialization cycles should not be included in the miss event count since those initialization cycles were filtered out with the init_hour config file option.

A few lines from the log file demonstrate the issue.  06Z and 18Z initializations should not be counted as miss events in this example.

```
DEBUG 4: GFS 20210823_060000 initialization, 120 lead, 20210828_060000 BEST track AL102021 genesis at (13.7, -49.8) is a dev and ops MISS.
DEBUG 4: GFS 20210823_120000 initialization, 114 lead, 20210828_060000 BEST track AL102021 genesis at (13.7, -49.8) is a dev and ops MISS.
DEBUG 4: GFS 20210823_180000 initialization, 108 lead, 20210828_060000 BEST track AL102021 genesis at (13.7, -49.8) is a dev and ops MISS.
DEBUG 4: GFS 20210824_000000 initialization, 102 lead, 20210828_060000 BEST track AL102021 genesis at (13.7, -49.8) is a dev and ops MISS.
DEBUG 4: GFS 20210824_060000 initialization, 96 lead, 20210828_060000 BEST track AL102021 genesis at (13.7, -49.8) is a dev and ops MISS.
DEBUG 4: GFS 20210824_120000 initialization, 90 lead, 20210828_060000 BEST track AL102021 genesis at (13.7, -49.8) is a dev and ops MISS.
DEBUG 4: GFS 20210824_180000 initialization, 84 lead, 20210828_060000 BEST track AL102021 genesis at (13.7, -49.8) is a dev and ops MISS.
```

The config file, log file with verbosity 4, and GENMPR output are attached for reference.
[filter_test.log](https://github.com/dtcenter/METplus/files/8577950/filter_test.log)
[filter_test_genmpr.txt](https://github.com/dtcenter/METplus/files/8577951/filter_test_genmpr.txt)
[TCGen_filter_config.txt](https://github.com/dtcenter/METplus/files/8577954/TCGen_filter_config.txt)

I'd suggest that this issue could be resolved in a bugfix release.

Thanks,
Dan

**Comment by JohnHalleyGotway** [comment, marked as answer]
@halperin-erau thanks for pointing out this issue. Yes, I definitely agree that tc_gen shouldn't be counting MISSES for any initialization hour that doesn't appear in the init_hour config file entry.

I was also able to independently verify this behavior by tweaking one of the calls in test/xml/unit_tc_gen.xml. I'll write up a GitHub bugfix issue for this and get to the bottom of it. Once I do, I'll make note of that issue here.

**Reply by JohnHalleyGotway** [reply]
@halperin-erau here's the GitHub issue for this bugfix: dtcenter/MET#2148

I'm going to mark this discussion as being answered and lock conversation on it. Let's do any future discussion about it as comments on that issue.

Thanks again for finding this!
